mllib/src/main/scala/org/apache/spark/ml/evaluation/Evaluator.scala

```diff
@@ -46,5 +46,12 @@ abstract class Evaluator extends Params {
    */
   def evaluate(dataset: DataFrame): Double
 
+  /**
+   * Indicates whether the metric returned by [[evaluate()]] should be maximized (true)
+   * or minimized (false).
+   * A given evaluator may support multiple metrics which may be maximized or minimized.
+   */
+  def shouldMaximize: Boolean
```

Comment:
* `Evaluator` reports the metrics but not maximizes them. I would suggest `isLargerBetter` instead.
* This is a break change if we don't put a default value. I would suggest a default value `true` to keep it consistent with implementations in Spark 1.4.